### DISCUSSION TITLE
Support const enum?

### DISCUSSION THREAD

**Discussion by Tofandel** [vitest-dev/vitest#3964] [General]
I have the same issue as #1650 which is locked (I really don't think it's helpful to lock old issues as this creates thread disparity and hard to find information and also hinders discussion) 

And the given reason is that vite transpiles to js without types, but I also tried `ts-mocha` which is also [transpile only by default](https://github.com/piotrwitek/ts-mocha/blob/master/src/index.js#L8C12-L8C19) and yet `const enum` works there, so what are they doing differently for this to work?

**Comment by sheremet-va** [comment]
Vite uses esbuild which doesn't support const enum.

**Reply by Tofandel** [reply]
See: https://github.com/evanw/esbuild/issues/128#issuecomment-848283225

Esbuild does support const enum, I just think it's because of `isolatedModules` of `vite` and `export declare const enum` instead of `export const enum`

So maybe we can remove the `declare` in the vue source or maybe add a vite plugin to transform this kind of enums to enums that can be understood by esbuild without types

**Reply by sheremet-va** [reply]
Vitest does no additional processing otherwise there will be a difference between dev and tests for people who use Vite.

If you want any changes to how files are transformed, you can start discussion in Vite repo.

**Comment by yangmingshan** [comment]
Vitest didn't support const enum, but it works fine with Vue which has many const enum, how did this happened? Do you have any clue? @sheremet-va

**Reply by yangmingshan** [reply]
I figure it out, esbuild can handle const enum, but it can't handle package's const enum, because the const enum only exist in the package's type file.

I opened a feature request in esbuild: https://github.com/evanw/esbuild/issues/3359, please upvote if you want it too.

**Comment by heyakyra** [comment]
Would `vue-tsc` help here? e.g. Set the `build` script in your component library's `package.json` to: `vue-tsc --declaration --emitDeclarationOnly && vite build`?

I found this workaround from: https://github.com/kulshekhar/ts-jest/issues/281#issuecomment-500780488

[yoasia](https://github.com/yoasia) commented [on Jun 11, 2019](https://github.com/kulshekhar/ts-jest/issues/281#issuecomment-500780488)

> I had the same problem. Here is what helped me:
> 
>     1. Make enum const.
>        `export const enum OneTwoThreeEnum { One = 'one', Two = 'two', Three = 'three' }`
> 
>     2. Use it like this:
>        `const something = "One" as OneTwoThreeEnum.One`

I haven't tried it, but someone else said this

[cbaron](https://github.com/cbaron) commented [on Feb 18, 2020](https://github.com/kulshekhar/ts-jest/issues/281#issuecomment-588020430)

> I found a workaround that sort of makes things okay:
> 
> It works if you have a module file that _only_ exports enums
> 
> ```
>      jest": "^24.9.0",
>     "ts-jest": "^24.2.0",
>     "typescript": "^3.7.2"
> ```
